toolboxTreppe/toolboxTreppe.ino

```diff
@@ -86,43 +71,49 @@ void loop()
 
 
 //-------------------------------Animationsfunktion ------------
-void animation1() {
-  currentMillis = millis();
+void animation1(uint8_t& y, uint8_t& x, uint8_t& gegenstrecke) {
+  static uint8_t xA;
+  static uint8_t xB;
+  static unsigned long xMillis;
+  static long onewayMillis;
+  static uint8_t hue;
+  
+  unsigned long currentMillis = millis();
   if (currentMillis - xMillis > xTime) {
     xMillis = currentMillis;
-    ++m_x;
-    if (m_x > 5) {
-      m_x = 0;
-      ++m_y;
+    ++x;
+    if (x > 5) {
+      x = 0;
+      ++y;
       //hue++;
       fadeToBlackBy(leds, NUM_LEDS, 20);
-      if (m_y > kMatrixHeight-1) {
-        m_y = 0;
+      if (y > kMatrixHeight-1) {
+        y = 0;
       }
     }
-    xA = 5 - m_x;
-    xB = 5 + m_x;
-    }
-    if (currentMillis - onewayMillis > onewayTime) {
-      onewayMillis = currentMillis;
-      if (gegenstrecke == 0) {
-        gegenstrecke = kMatrixHeight-1;
-      } else {
-        --gegenstrecke;
-      }
+    xA = 5 - x;
+    xB = 5 + x;
+  }
+  if (currentMillis - onewayMillis > onewayTime) {
+    onewayMillis = currentMillis;
+    if (gegenstrecke == 0) {
+      gegenstrecke = kMatrixHeight-1;
+    } else {
+      --gegenstrecke;
     }
+  }
     
-    ++hue;
-    leds[ XY(xA, m_y)] = CHSV( hue, 255, 255 );
-    leds[ XY(xB, m_y)] = CHSV( hue, 255, 255 );
-    leds[ XY(5, gegenstrecke)] = CHSV( hue, 255, 255 );
-    FastLED.show();
+  ++hue;
+  leds[ XY(xA, y)] = CHSV( hue, 255, 255 );
+  leds[ XY(xB, y)] = CHSV( hue, 255, 255 );
+  leds[ XY(5, gegenstrecke)] = CHSV( hue, 255, 255 );
+  FastLED.show();
 }
 //----------------------------- Debug Funktion ----------------
-void debug() {
-  Serial.print(m_y);
+void debug(const uint8_t& y, const uint8_t& x, const uint8_t& gegenstrecke) {
```

Comment:
debug could be a generic print function like sprintf

Reply:
the debug function already makes no sence in the current way anyways.
He activates debug when pin1 and 2 are both high and then in the debugfunction 
so the debug output Oben and Unten can´t appear and the output Mitte will always appear

Reply:
it cleanup time

Reply:
I wrote the debug function before i planned how to acces it and then i forgot to remove the section with the button modes...